libs/prelude/Prelude/Algebra.idr

```diff
@@ -286,44 +160,19 @@ class (VerifiedBoundedJoinSemilattice a, VerifiedBoundedMeetSemilattice a, Verif
 ||| + Unity for `<.>`:
 |||     forall a,     a <.> unity     == a
 |||     forall a,     unity <.> a     == a
-||| + InverseM of `<.>`:
-|||     forall a,     a <.> inverseM a == unity
-|||     forall a,     inverseM a <.> a == unity
+||| + InverseM of `<.>`, except for neutral
+|||     forall a /= neutral,  a <.> inverseM a == unity
+|||     forall a /= neutral,  inverseM a <.> a == unity
 ||| + Distributivity of `<.>` and `<->`:
 |||     forall a b c, a <.> (b <+> c) == (a <.> b) <+> (a <.> c)
 |||     forall a b c, (a <+> b) <.> c == (a <.> c) <+> (b <.> c)
-class RingWithUnity a => Field a where
-  inverseM : a -> a
+class (RingWithUnity a, Eq a) => Field a where
+  inverseM : (x : a) -> (notId : x == neutral = False) -> a
```

Comment:
1. We can simplify `(notId : x == neutral = False)` into `(notZero: not (x == neutral))` or even `(x /= neutral)`.
2. ...or rather we can use propositional equality `(Not (x = neutral))` to avoid dependency on `Eq a`. What is better?